mlfinlab/structural_breaks/sadf.py

```diff
@@ -10,13 +10,15 @@
 
 # pylint: disable=invalid-name
 
-def _get_sadf_at_t(X: pd.DataFrame, y: pd.DataFrame, min_length: int) -> float:
+def _get_sadf_at_t(X: pd.DataFrame, y: pd.DataFrame, min_length: int, model: str, phi: float) -> float:
     """
     Snippet 17.2, page 258. SADF's Inner Loop (get sadf value at t)
 
     :param X: (pd.DataFrame) of lagged values, constants, trend coefficients
     :param y: (pd.DataFrame) of y values (either y or y.diff())
     :param min_length: (int) minimum number of samples needed for estimation
+    :param model: (str) either 'linear', 'quadratic', 'sm_poly_1', 'sm_poly_2', 'sm_exp', 'sm_power'
+    :param phi: (float) in [0, 1], penalize large sample lengths when computing SMT
```

Comment:
Is this a form of regularisation?

Reply:
Yes, the method as it stands now `(phi=0)` is biased towards finding long-run bubbles. If we are interested in finding also short-run bubbles, we should penalize large sample lengths.